[llvm-debuginfo-analyzer][DOC] Convert 'README.txt' to markdown.

CarlosAlbertoEnciso opened the pull request:
As part of the WebAssembly support work
https://github.com/llvm/llvm-project/pull/85566

The README.txt is a bit odd since it only lists issues and problems without talking about what works. It’s also hard to read on the GitHub web view.

- Convert to Markdown and linking to the command docs https://llvm.org/docs/CommandGuide/llvm-debuginfo-analyzer
- Rename some left 'elf reader' to 'DWARF reader'.

CarlosAlbertoEnciso committed:
[llvm-debuginfo-analyzer][DOC] Convert 'README.txt' to markdown.

As part of the WebAssembly support work
https://github.com/llvm/llvm-project/pull/85566

The README.txt is a bit odd since it only lists issues and
problems without talking about what works. It’s also hard
to read on the GitHub web view.

- Convert to Markdown and linking to the command docs
  https://llvm.org/docs/CommandGuide/llvm-debuginfo-analyzer
- Rename some left 'elf reader' to 'DWARF reader'.

CarlosAlbertoEnciso commented:
> ⚠️ We detected that you are using a GitHub private e-mail address to contribute to the repo. Please turn off [Keep my email addresses private](https://github.com/settings/emails) setting in your account. See [LLVM Discourse](https://discourse.llvm.org/t/hidden-emails-on-github-should-we-do-something-about-it) for more information.

Fixed.

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,171 @@
+# llvm-debuginfo-analyzer
```
```suggestion
# `llvm-debuginfo-analyzer`
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,171 @@
+These are the notes collected during the development, review and test.
+They describe limitations, know issues and future work.
+
+### Remove the use of macros in *LVReader.h* that describe the *bumpallocators*.
```
Consider monospace / code style (ticks) instead of emphasis (asterisks) for code things like files and functions

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,171 @@
+# llvm-debuginfo-analyzer
+
+These are the notes collected during the development, review and test.
+They describe limitations, know issues and future work.
```
```suggestion
They describe limitations, known issues and future work.
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,171 @@
+One specific case is the **LVSymbolNames** definitions.
+
+### Calculate unique offset for *CodeView* elements.
+In order to have the same logical functionality as the **DWARF Reader**, such
```
```suggestion
In order to have the same logical functionality as the **DWARF reader**, such
```

jryans reviewed (approved):
Looks good overall, thanks! 😄 I noted a few optional stylistic points for consideration.

CarlosAlbertoEnciso committed:
[llvm-debuginfo-analyzer][DOC] Convert 'README.txt' to markdown.

As part of the WebAssembly support work
https://github.com/llvm/llvm-project/pull/85566

The README.txt is a bit odd since it only lists issues and
problems without talking about what works. It’s also hard
to read on the GitHub web view.

- Use monospace instead of emphasis style.

CarlosAlbertoEnciso commented:
@jryans Thanks for your review.

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+### Add support for some extra `DWARF` locations.
+The following ``DWARF`` debug location operands are not supported:
+
+* DW_OP_const_type
+* DW_OP_entry_value
+* DW_OP_implicit_value
```
```suggestion
* `DW_OP_const_type`
* `DW_OP_entry_value`
* `DW_OP_implicit_value`
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+* DW_OP_implicit_value
+
+### Add support for additional binary formats.
+* Extended COFF ``(XCOFF)``
```
```suggestion
* Extended COFF (`XCOFF`)
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+to expose equality and hasher) for the allocators and the creation
+functions could be a function template.
+
+### Use a ``lit test`` instead of a ``unit test`` for the ``logical readers``.
```
```suggestion
### Use a **lit test** instead of a **unit test** for the **logical readers**.
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+
+As the ``DebugInfoLogicalView`` library is sufficiently exposed via the
+``llvm-debuginfo-analyzer`` tool, follow the LLVM general approach and
+use ``LIT`` tests to validate the ``logical readers``.
```
```suggestion
use ``lit`` tests to validate the **logical readers**.
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+### Remove ``LVScope::Children`` container.
+**[D137933](https://reviews.llvm.org/D137933#inline-1373902)**
+
+Use a ``chaining iterator`` over the other containers rather than keep a
```
```suggestion
Use a **chaining iterator** over the other containers rather than keep a
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+The more idiomatic LLVM way to handle this would be to have ``find``
+return ``Optional<ValueType>``.
+
+### Pass references instead of pointers (``Comparison functions``).
```
```suggestion
### Pass references instead of pointers (**Comparison functions**).
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+### Pass references instead of pointers (``Comparison functions``).
+**[D125782](https://reviews.llvm.org/D125782#inline-1293920)**
+
+In the ``comparison functions``, pass references instead of pointers (when
```
```suggestion
In the **comparison functions**, pass references instead of pointers (when
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+[D137156](https://reviews.llvm.org/D137156)**
+
+During the traversal of the debug information sections, we created the
+logical lines representing the ``disassembled instructions`` from the ``text
+section`` and the logical lines representing the ``line records`` from the
+``debug line`` section. Using the ranges associated with the logical scopes,
```
```suggestion
logical lines representing the **disassembled instructions** from the **text
section** and the logical lines representing the **line records** from the
**debug line** section. Using the ranges associated with the logical scopes,
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+Only linked executables are handled. It does not support relocatable
+files compiled with ``-ffunction-sections``.
+
+### Add support for DWARF v5 .debug_names section / CodeView public symbols stream.
```
```suggestion
### Add support for DWARF v5 `.debug_names` section / CodeView public symbols stream.
```

jryans commented on llvm/tools/llvm-debuginfo-analyzer/README.md:
```diff
@@ -0,0 +1,170 @@
+
+The ``DWARF`` and ``CodeView`` readers use the public names information to create
+the instructions (``LVLineAssembler``). Instead of relying on ``DWARF`` section
+names (``.debug_pubnames``, ``.debug_names``) and ``CodeView`` ``public symbol`` stream
```
```suggestion
names (``.debug_pubnames``, ``.debug_names``) and ``CodeView`` public symbol stream
```

jryans reviewed (approved):
Thanks! I added some additional stylistic comments. They are all optional though, so apply what makes sense to you.

About the words "DWARF", "CodeView", and "PDB", I wasn't sure whether to treat them as "code" (and put in monospace) or just leave them alone without styling... Anyway, probably best to choose one approach for such words and use it consistently throughout.

dwblaikie commented:
> About the words "DWARF", "CodeView", and "PDB", I wasn't sure whether to treat them as "code" (and put in monospace) or just leave them alone without styling... Anyway, probably best to choose one approach for such words and use it consistently throughout.

In general I'd say all three appear often enough and refer to a concept, that I wouldn't treat them as code identifiers - keep them as plain english language identifiers. (eg: I think it'd be weird if the DWARF spec used monospaced for every appearance of the word 'DWARF' in the spec)

CarlosAlbertoEnciso committed:
[llvm-debuginfo-analyzer][DOC] Convert 'README.txt' to markdown.

As part of the WebAssembly support work
https://github.com/llvm/llvm-project/pull/85566

The README.txt is a bit odd since it only lists issues and
problems without talking about what works. It’s also hard
to read on the GitHub web view.

- Address more feedback in relation to the correct Use
  of monospace.
- Keep DWARF, CodeView, PDB as plain text.

CarlosAlbertoEnciso commented:
@jryans @dwblaikie Thanks very much for your additional reviews.
